lib/httpserver/httpserver.go

```diff
@@ -396,6 +396,15 @@ func handlerWrapper(s *server, w http.ResponseWriter, r *http.Request, rh Reques
 		// See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4128
 		fmt.Fprintf(w, "User-agent: *\nDisallow: /\n")
 		return
+	case "/config", "/-/reload":
+		w = &responseWriterWithAbort{
+			ResponseWriter: w,
+		}
+		if !rh(w, r) {
```

Comment:
Looks like it breaks backward compatibility.
Before:
1. User sets BA
2. User doesn't set -reloadAuthKey
3. Request to `/-/reload` will check the BA

Now:
1. User sets BA
2. User doesn't set -reloadAuthKey
3. Request to `/-/reload` will **not** check the BA

This change may make `/config` and `/-/reload` endpoints open for users who relied on `httpAuth.*` only.
It also acts differently to `-metricsAuthKey`, `-flagsAuthKey`, etc.

Reply:
Thanks for the explanation!